.github/workflows/tests.yml

```diff
@@ -25,17 +30,17 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.6", "3.7", "3.8", "3.9", "3.10"]
+        python-version: ["3.6", "3.7", "3.8", "3.9", "3.10", "3.11"]
         os: [ubuntu-latest, macos-latest, windows-latest]
 
     steps:
     - name: Checkout
-      uses: actions/checkout@v2
+      uses: actions/checkout@v3
 
     - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v2
+      uses: actions/setup-python@v4
       with:
-        python-version: ${{ matrix.python-version }}
+        python-version: ${{ matrix.python-version }}-dev
```

Comment:
Do we want to test against the dev version of all Python versions?

Reply:
Yes. Apparently the dev versions are the same as the non-dev versions except they allow to get pre-releases prior to the first final release.

Reply:
[ref](https://github.com/actions/setup-python/issues/213#issuecomment-1191293053)

Reply:
I read the reference, but TBH I think using `-dev` for this is misleading 🙁 I'd rather we only used `-dev` for places where we really do want the development version of Python (alphas, betas, RCs).

I really don't understand the logic from the setup-python developers here.

Reply:
A slightly more direct reference: https://github.com/actions/setup-python#available-versions-of-python

> Use -dev instead of a patch number (e.g., 3.11-dev) to install the latest patch version release for a given minor version, alpha and beta releases included.